torch/distributions/gamma.py

```diff
@@ -70,6 +70,11 @@ def log_prob(self, value):
                 (self.concentration - 1) * torch.log(value) -
                 self.rate * value - torch.lgamma(self.concentration))
 
+    def cdf(self, value):
+        if self._validate_args:
+            self._validate_sample(value)
+        return torch.igamma(self.concentration, self.rate * value) # depsite its name, torch.igamma is the regularized gamma function
```

Comment:
cc @kshitij12345 to also take a look. 

What name did you expect this function to have, @jan-glx? We're implementing the torch.special module now with updated names.

Reply:
Apparently there is quite some inconsistency across fields/programming languages:

| language | upper (Γ) - and |lower incomplete gamma function (γ) | regularized lower gamma function (P=γ/Γ) | 
| ---- | -----| ------| ------|
|[Wikipedia](https://en.wikipedia.org/wiki/Incomplete_gamma_function#Regularized_Gamma_functions_and_Poisson_random_variables)| upper incomplete gamma function | lower incomplete gamma function | regularized gamma function |
|[dlmf](https://dlmf.nist.gov/8.2#E4) | incomplete gamma function γ|  incomplete gamma function Γ | Normalized incomplete gamma function| 
| [julia](https://juliamath.github.io/SpecialFunctions.jl/dev/functions_list/#SpecialFunctions.gamma_inc) | gamma(a,x)|       | gamma_inc(a,x)[1] | 
| [scipy](https://docs.scipy.org/doc/scipy/reference/special.html#gamma-and-related-functions) |  |  | gammainc(a, x) |
| [GNU SF](https://www.gnu.org/software/gsl/doc/html/specfunc.html?highlight=gamma#incomplete-gamma-functions)| gsl_sf_gamma_inc | | gsl_sf_gamma_inc_P |  
| [Wolfram](https://reference.wolfram.com/language/tutorial/MathematicalFunctions.html#21909) | Gamma[a, z]|  Gamma[a,0,z1] | GammaRegularized |
| R |  |  | pgamma(z, a) |
| [TF](https://www.tensorflow.org/api_docs/python/tf/math/igamma) | | | tf.math.igamma |
| [MATLAB](https://de.mathworks.com/help/symbolic/sym.igamma.html#bt6_p8p-1) |  gammainc(x, a, 'lower')| igamma(a, x) or gammainc(x, a, 'upper') | gammainc (x,a,scale='scaledlower' )|

Not sure what is best, but igamma could also mean inverse gamma, so prob. not ideal.
While not indicating the normalization, id probably just stick with scipy....

Reply:
Thanks for the detailed explanation.

> Not sure what is best, but igamma could also mean inverse gamma, so prob. not ideal.

I understand your concern now. As @mruberry mentioned, the special module will update the name of `igamma` to being consistent with SciPy's (i.e. `gammainc`)

Reference to the tracker issue: https://github.com/pytorch/pytorch/issues/50345

Reply:
Thank you for sharing your perspective, @jan-glx.

Reply:
(removed the comment now, otherwise it will probably be forgotten after the rename of the specials)